extensions/numi/src/services/requests.tsx

```diff
@@ -31,3 +36,16 @@ export const query = (q?: string): Promise<string[]> => {
     }
   });
 };
+
+export const queryWithNumiCli = async (q?: string, path?: string): Promise<string[]> => {
+  const command = `${path || DefaultPreferences.numi_cli_binary_path} "${q}"`;
```

Comment:
Always use `DefaultPreferences.numi_cli_binary_path`; if there is any security flaw that permits the call of this function, modifying the path parameter could handle a big problem if the user has failures. **They need to set up the correct path on the preferences.**